frameworks/hdfs/alerts/HDFS_alert_rules.yml

```diff
@@ -0,0 +1,105 @@
+
+  - name: hdfs_warning
+    rules:
+    - alert: CapacityRemainingGB
+      expr: sum by (service_name,task_name) ({__name__=~".*_FSNamesystem_CapacityRemainingGB"}) < 10
+      for: 10m
+      labels:
+       severity: warning
+       service: hdfs
+      annotations:
+       summary: "Low capacity remaining in GB on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has capacity remaining {{ $value }}GB"
+
+    - alert: MissingBlocks
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_MissingBlocks"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Missing blocks on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} missed blocks equals to {{ $value }}"
+
+    - alert: UnderReplicatedBlocks
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_UnderReplicatedBlocks"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Under replicated blocks on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has under replicated blocks equals to {{ $value }}"
+
+    - alert: NumStaleDataNodes
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_StaleDataNodes"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Number of stale DataNodes on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has number stale DataNodes equals to {{ $value }}"
+
+    - alert: MemHeapUsed
+      expr: (sum({__name__=~".*_jvm_JvmMetrics_MemHeapMaxM"}) / sum({__name__=~".*_jvm_JvmMetrics_MemHeapUsedM"})) < 0.8
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Total service memory heap in used for {{ $labels.instance }}"
+       description: "{{ $labels.instance }} has heap in used greater then 0.8 (current value: {{ $value }})"
+
+    - alert: MemUsedPercent
+      expr: mem_used_percent > 80
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Instance memory in used percent for {{ $labels.instance }}"
+       description: "{{ $labels.instance }} has memory in used greater then 80% (current value: {{ $value }}%)"
+
+  - name: hdfs_critical
+    rules:
+    - alert: InstanceDown
+      expr: up == 0
```

Comment:
I realized that this metric is a DCOS metric and represents how many agents are `up` and running. It doesn't have anything to do with HDFS and we should remove it.

However, we should alert on any nodes going down for the HDFS service. I tried to look for some native HDFS metrics specific to that but couldn't find any but I think your metric below with `numAllSource - numActiveSources` is the pretty much the same thing.